<?xml version="1.0"?>
<launch>
  <node pkg="ig_active_reconstruction_octomap" type="octomap_world_representation" name="octomap_world_representation" clear_params="true" output="screen">
    
    <!--Octree configuration-->
    <param name="resolution_m" value="0.01" />
    <param name="occupancy_threshold" value="0.5" />
    <param name="hit_probability" value="0.7" />
    <param name="miss_probability" value="0.4" />
    <param name="clamping_threshold_min" value="0.12" />
    <param name="clamping_threshold_max" value="0.97" />
    
    <!-- PCL input configuration -->
    <param name="world_frame_name" value="world" />
    <param name="use_bounding_box" value="true" />
    <param name="bounding_box_min_point_m/x" value="-0.6" />
    <param name="bounding_box_min_point_m/y" value="-0.6" />
    <param name="bounding_box_min_point_m/z" value="-0.01" />
    <param name="bounding_box_max_point_m/x" value="0.6" />
    <param name="bounding_box_max_point_m/y" value="0.6" />
    <param name="bounding_box_max_point_m/z" value="0.6" />
    <param name="max_sensor_range_m" value="1.5" />
    
    <!-- Occlusion calculation configuration -->
    <param name="occlusion_update_dist_m" value="0.3" />
    
    <!-- Raycaster configuration -->
    <param name="img_width_px" value="480" />
    <param name="img_height_px" value="752" />
    <param name="camera/fx" value="448.1008985853343" />
    <param name="camera/fy" value="448.1008985853343" />
    <param name="camera/cx" value="376.5" />
    <param name="camera/cy" value="240.5" />
    <param name="max_ray_depth_m" value="1.5" />
    
    <param name="raycasting/resolution_x" value="0.1" />
    <param name="raycasting/resolution_y" value="0.1" />
    <param name="raycasting/min_x_perc" value="0.25" />
    <param name="raycasting/min_y_perc" value="0.25" />
    <param name="raycasting/max_x_perc" value="0.75" />
    <param name="raycasting/max_y_perc" value="0.75" />
    
    <!-- Information gain config -->
    <param name="ig/p_unknown_prior" value="0.5" />
    <param name="ig/p_unknown_upper_bound" value="0.8" />
    <param name="ig/p_unknown_lower_bound" value="0.2" />
    <param name="ig/voxels_in_void_ray" value="100" />
    
  </node>
</launch>